pkg/apiserver/apiserver.go

```diff
@@ -254,8 +254,8 @@ func AddApiWebService(container *restful.Container, apiPrefix string, versions [
 }
 
 // Adds a service to return the supported api versions at /apis.
-func AddApisWebService(container *restful.Container, apiPrefix string, groups []unversioned.APIGroup) {
-	rootAPIHandler := RootAPIHandler(groups)
+func AddApisWebService(container *restful.Container, apiPrefix string, f func() []unversioned.APIGroup) {
```

Comment:
This change takes us from a static list to one that might change in between requests.  Is that intentional?  Should it always return a fixed value?  Could we not have resolved this by figuring out what third party resources were available before calling this method?  I don't think that list changes after initialization.